test/e2e/usersignup_test.go

```diff
@@ -69,64 +74,92 @@ func (s *userSignupIntegrationTest) TestAutomaticApproval() {
 
 		t.Run("reset the threshold and expect the user will be provisioned", func(t *testing.T) {
 			// when
-			hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().Enabled(true).ResourceCapacityThreshold(80))
+			hostAwait.UpdateToolchainConfig(
+				testconfig.AutomaticApproval().Enabled(true),
+				testconfig.CapacityThresholds().ResourceCapacityThreshold(80),
+			)
 
 			// then
 			userSignup, err := hostAwait.WaitForUserSignup(userSignup.Name,
 				wait.UntilUserSignupHasConditions(ConditionSet(Default(), ApprovedAutomatically())...),
 				wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValueApproved))
 			require.NoError(s.T(), err)
 			VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, userSignup, "deactivate30", "base")
+			// delete the userSignup
+			err = hostAwait.Client.Delete(context.TODO(), userSignup)
+			require.NoError(t, err)
+			err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(userSignup.GetName())
+			require.NoError(t, err)
```

Comment:
this should be done by the automatic cleanup
```suggestion
```

Reply:
Now I see that it's actually provisioned in the root-test and not in the sub-test. 
Then, instead of manually deleting it, I would probably shuffle around with the structure of the tests, so the tests are properly isolated and cleanup is done automaitcally. WDYT?